src/LMS1xx.cpp

```diff
@@ -117,9 +118,24 @@ status_t LMS1xx::queryStatus() {
 
 void LMS1xx::login() {
 	char buf[100];
+	int result;
 	sprintf(buf, "%c%s%c", 0x02, "sMN SetAccessMode 03 F4724744", 0x03);
+	
+	fd_set readset;
+	struct timeval timeout;
+	timeout.tv_sec = 1;
+	timeout.tv_usec = 0;
 
-	write(sockDesc, buf, strlen(buf));
+	do { //loop until data is available to read
+	  ros::Duration(1.0).sleep();
+
+	  write(sockDesc, buf, strlen(buf));
+
+	  FD_ZERO(&readset);
+	  FD_SET(sockDesc, &readset);
+	  result = select(sockDesc + 1, &readset, NULL, NULL, &timeout);
```

Comment:
It would loop forever if there is no response from the sensor. Would it be better to exit the loop after some arbitrary amount of time and then reconnect? And yes, the timeout.tv_sec/usec should be inside the loop.